examples/gpt/deepspeed_hf.py

```diff
@@ -165,12 +180,23 @@ def loss_fn(outputs, labels):
         model = model.to(device)
     report_memory(msg="After building model")
 
-    random_seed = 2000 + dist.get_rank()
+    if args.disable_pipeline or args.sequence_parallel:
+        random_seed = 2013 + dist.get_rank()
+    else:
+        random_seed = (
+            2013
+            + 100 * model.mpu.get_pipe_parallel_rank()
```

Comment:
Where does `.mpu` come from? Is this a DeepSpeed engine API?

Reply:
yeah, this is a deepspeed api, since model is a deepspeed object after pipeline build